core/src/style/styleMixer.cpp

```diff
@@ -0,0 +1,222 @@
+#include "styleMixer.h"
+#include "style/style.h"
+#include "util/topologicalSort.h"
+#include "yaml-cpp/yaml.h"
+
+#include <algorithm>
+#include <set>
+
+namespace Tangram {
+
+std::vector<std::string> StyleMixer::getStylesToMix(const Node& _style) {
+
+    std::vector<std::string> names;
+
+    // 'base' style is the first item to mix.
+    if (const Node& base = _style["base"]) {
+        if (base.IsScalar()) { names.push_back(base.Scalar()); }
+    }
+
+    // 'mix' styles are mixed next, in order of declaration.
+    if (const Node& mix = _style["mix"]) {
+        if (mix.IsScalar()) {
+            names.push_back(mix.Scalar());
+        } else if (mix.IsSequence()) {
+            for (const auto& m : mix) {
+                if (m.IsScalar()) { names.push_back(m.Scalar()); }
+            }
+        }
+    }
+
+    return names;
+}
+
+std::vector<std::string> StyleMixer::getMixingOrder(const Node& _styles) {
+
+    // Input must be a map of names to style configuration nodes.
+    if (!_styles.IsMap()) {
+        return {};
+    }
+
+    // Dependencies are pairs of strings that form a DAG.
+    // If style 'a' mixes style 'b', the dependency would be {'b', 'a'}.
+    std::vector<std::pair<std::string, std::string>> dependencies;
```

Comment:
I think for this allocation and the one you mentioned before, I could make them instance variables and clear them when appropriate. `StyleMixer` doesn't actually have any member variables right now so that would also help justify it as a class instead of just a namespace of functions.